CovertInfiltration/Src/CovertInfiltration/Classes/X2Helper_Infiltration.uc

```diff
@@ -1500,3 +1500,83 @@ static function UpdateFacilityMissionLocks (optional XComGameState NewGameState)
 //	}
 }
 
+static function bool UnitHasIrrelevantItems (StateObjectReference UnitRef)
+{
+	local XComGameStateHistory             History;
+	local XComGameState_Unit               UnitState;
+	local array<XComGameState_Item>        CurrentInventory;
+	local XComGameState_Item               InventoryItem;
+	local X2InfiltrationModTemplateManager InfilTemplateManager;
+	local X2InfiltrationModTemplate        InfilTemplate;
+
+	if (UnitRef.ObjectID <= 0) return false;
+	
+	History = `XCOMHISTORY;
+	UnitState = XComGameState_Unit(History.GetGameStateForObjectID(UnitRef.ObjectID));
+	InfilTemplateManager = class'X2InfiltrationModTemplateManager'.static.GetInfilTemplateManager();
+
+	// loop through items
+	CurrentInventory = UnitState.GetAllInventoryItems();
+	foreach CurrentInventory(InventoryItem)
+	{
+		if (InventoryItem.GetMyTemplate().bInfiniteItem) continue;
+
+		// check item
+		InfilTemplate = InfilTemplateManager.GetInfilTemplateFromItem(InventoryItem.GetMyTemplate());
+
+		if (InfilTemplate != none)
+		{
+			if (InfilTemplate.HoursAdded >= 0 && InfilTemplate.Deterrence <= 0)
+			{
+				return true;
+			}
+		}
+		else
+		{
+			// check category
+			InfilTemplate = InfilTemplateManager.GetInfilTemplateFromCategory(InventoryItem.GetMyTemplate());
+			
+			if (InfilTemplate != none)
+			{
+				if (InfilTemplate.HoursAdded >= 0 && InfilTemplate.Deterrence <= 0)
+				{
+					return true;
+				}
+			}
+			else
+			{
+				return true;
```

Comment:
You are breaking the loop here (skipping next items) - you sure you want to do that?

Reply:
Yes, if the soldier has one irrelevant item then it doesn’t matter what the status of the rest of the items are, so the function returns true.